onc-healthit/lantern-back-end
# Populate validation

To test this PR:
- Test the migration script again since it's been updated so that for a specific URL, the fhir_endpoints_info row has the same validation_result_id as the most recent row in the history table for that URL.
- Test that when running the whole system, the validation_result_id is updated whenever the fhir_endpoints_info table is updated
- There should no longer be an HTTPResponse validation for all validations, and for 4.0 validations the smart http response validation should instead be a check that the smart response exists.

Running with the migration script instructions

- Follow the usual migration steps
- One the database has been migrated, `make run`
- Kill the endpoint manager: `docker stop lantern-back-end_endpoint_manager_1`
- To migrate the validation information into the validation table, from `capabilityreceiver/cmd/migratevalidations`, run `go run main.go up`
- You can run `go run main.go down` to do a down migration (putting the validation information into the validation field)
- Check that:
-- all rows (for both tables) have a populated validation_result_id field
-- the validations table is populated correctly
-- the validation_result_id field is also populated

Pull requests into this repository require the following checks to be completed by the submitter and reviewers.

**Submitter:**
  - This pull request describes why these changes were made.
  - The JIRA ticket links to this PR.
  - The JIRA ticket for this PR is: https://oncprojectracking.healthit.gov/support/browse/LANTERN-329
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.
  - Code diff has been done and checked to ensure only expected code is being committed.
  - A linter for the language has been run.

**Primary Reviewer:**

  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
  - Code is maintainable, reusable, efficient, and correct.
  - Code accomplishes the tasks purpose.
  - Code follows style guidance appropriate for the language, including passing linter checks.
  - Code is well commented (which does not mean verbosely commented).
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.

**Secondary Reviewer:**

  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
  - Code accomplishes the tasks purpose.
  - Tests have been run locally and pass.

**Merger:**

Prior to merging, ensure that all checks have been completed. Delete the branch after the merge
is complete.

Create a new branch `<branch name>_gomodupdate`. Run `make update_mods` to update the `go.mod`
and `go.sum` files to point to master. Create a PR. Require one sanity check reviewer.
